### ISSUE TITLE
GStreamer gst-plugin-scan coredumps with libcamera.so.0.1.0

### ISSUE THREAD

**Issue by alexzeitgeist** [pipewire-debian/pipewire-debian#33]
**Describe the bug**

Upgraded to pipewire 0.3.37-1~ubuntu21.04 on hirsute. Upon reboot, GStreamer produces a coredump on libcamera.so.0.1.0.

**Install logs**

```bash
2021-09-25 19:18:02 upgrade pipewire-media-session:amd64 0.3.36-1~ubuntu21.04 0.3.37-1~ubuntu21.04
2021-09-25 19:18:02 upgrade pipewire-pulse:amd64 0.3.36-1~ubuntu21.04 0.3.37-1~ubuntu21.04
2021-09-25 19:18:02 upgrade gstreamer1.0-pipewire:amd64 0.3.36-1~ubuntu21.04 0.3.37-1~ubuntu21.04
2021-09-25 19:18:03 upgrade pipewire-tests:amd64 0.3.36-1~ubuntu21.04 0.3.37-1~ubuntu21.04
2021-09-25 19:18:03 upgrade pipewire-bin:amd64 0.3.36-1~ubuntu21.04 0.3.37-1~ubuntu21.04
2021-09-25 19:18:03 upgrade pipewire-audio-client-libraries:amd64 0.3.36-1~ubuntu21.04 0.3.37-1~ubuntu21.04
2021-09-25 19:18:04 upgrade libpipewire-0.3-dev:amd64 0.3.36-1~ubuntu21.04 0.3.37-1~ubuntu21.04
2021-09-25 19:18:04 upgrade libpipewire-0.3-0:amd64 0.3.36-1~ubuntu21.04 0.3.37-1~ubuntu21.04
2021-09-25 19:18:04 upgrade libspa-0.2-jack:amd64 0.3.36-1~ubuntu21.04 0.3.37-1~ubuntu21.04
2021-09-25 19:18:04 upgrade libspa-0.2-bluetooth:amd64 0.3.36-1~ubuntu21.04 0.3.37-1~ubuntu21.04
2021-09-25 19:18:05 upgrade libspa-0.2-modules:amd64 0.3.36-1~ubuntu21.04 0.3.37-1~ubuntu21.04
2021-09-25 19:18:05 upgrade pipewire:amd64 0.3.36-1~ubuntu21.04 0.3.37-1~ubuntu21.04
2021-09-25 19:18:05 upgrade libpipewire-0.3-modules:amd64 0.3.36-1~ubuntu21.04 0.3.37-1~ubuntu21.04
2021-09-25 19:18:05 upgrade libspa-0.2-dev:amd64 0.3.36-1~ubuntu21.04 0.3.37-1~ubuntu21.04
2021-09-25 19:18:06 install liburcu6:amd64 <none> 0.12.2-1build1
2021-09-25 19:18:06 install liblttng-ust-ctl4:amd64 <none> 2.12.1-1ubuntu1
2021-09-25 19:18:06 install liblttng-ust0:amd64 <none> 2.12.1-1ubuntu1
2021-09-25 19:18:06 install libcamera0:amd64 <none> 0.r3055.40f5fddc-2~ubuntu21.04
2021-09-25 19:18:06 upgrade pipewire-doc:all 0.3.36-1~ubuntu21.04 0.3.37-1~ubuntu21.04
2021-09-25 19:18:12 upgrade pipewire-locales:all 0.3.36-1~ubuntu21.04 0.3.37-1~ubuntu21.04
```

**Reboot logs**

```bash
Sep 26 09:49:25 thinkpad-linux kernel: traps: gst-plugin-scan[2788] trap divide error ip:7ffa4d09c3cb sp:7ffee632d600 error:0 in libcamera.so.0.1.0[7ffa4d04d000+c3000]
Sep 26 09:49:25 thinkpad-linux systemd[1]: Created slice system-systemd\x2dcoredump.slice.
Sep 26 09:49:25 thinkpad-linux systemd[1]: Started Process Core Dump (PID 2827/UID 0).
Sep 26 09:49:25 thinkpad-linux systemd-coredump[2828]: Process 2788 (gst-plugin-scan) of user 1000 dumped core.
                                                       
                                                       Stack trace of thread 2788:
                                                       #0  0x00007ffa4d09c3cb _ZN9libcamera14ControlInfoMap8validateEv (libcamera.so.0 + 0x8a3cb)
                                                       #1  0x00007ffa4d09c5b4 _ZN9libcamera14ControlInfoMapC1ESt16initializer_listISt4pairIKPKNS_9ControlIdENS_11ControlInfoEEERKSt13unordered_mapIjS5_St4hashIjESt8equal_toIjESaIS2_IKjS5_EEE (libcamera.so.0 + 0x8a5b4)
                                                       #2  0x00007ffa4d061230 n/a (libcamera.so.0 + 0x4f230)
                                                       #3  0x00007ffa4d065014 n/a (libcamera.so.0 + 0x53014)
                                                       #4  0x00007ffa51a3e96e call_init (ld-linux-x86-64.so.2 + 0x1196e)
                                                       #5  0x00007ffa51a3ea58 call_init (ld-linux-x86-64.so.2 + 0x11a58)
                                                       #6  0x00007ffa517022e5 __GI__dl_catch_exception (libc.so.6 + 0x1582e5)
                                                       #7  0x00007ffa51a4311a dl_open_worker (ld-linux-x86-64.so.2 + 0x1611a)
                                                       #8  0x00007ffa51702288 __GI__dl_catch_exception (libc.so.6 + 0x158288)
                                                       #9  0x00007ffa51a426ed _dl_open (ld-linux-x86-64.so.2 + 0x156ed)
                                                       #10 0x00007ffa513f434c dlopen_doit (libdl.so.2 + 0x234c)
                                                       #11 0x00007ffa51702288 __GI__dl_catch_exception (libc.so.6 + 0x158288)
                                                       #12 0x00007ffa51702353 __GI__dl_catch_error (libc.so.6 + 0x158353)
                                                       #13 0x00007ffa513f4b89 _dlerror_run (libdl.so.2 + 0x2b89)
                                                       #14 0x00007ffa513f43d8 __dlopen (libdl.so.2 + 0x23d8)
                                                       #15 0x00007ffa5154abdf g_module_open (libgmodule-2.0.so.0 + 0x1bdf)
                                                       #16 0x00007ffa51968d67 n/a (libgstreamer-1.0.so.0 + 0x9cd67)
                                                       #17 0x00007ffa5196a6d2 n/a (libgstreamer-1.0.so.0 + 0x9e6d2)
                                                       #18 0x00007ffa5196b648 _gst_plugin_loader_client_run (libgstreamer-1.0.so.0 + 0x9f648)
                                                       #19 0x0000564160c011dc n/a (gst-plugin-scanner + 0x11dc)
                                                       #20 0x00007ffa515d2565 __libc_start_main (libc.so.6 + 0x28565)
                                                       #21 0x0000564160c0123e _start (gst-plugin-scanner + 0x123e)
                                                       
                                                       Stack trace of thread 2826:
                                                       #0  0x00007ffa516bbf6d syscall (libc.so.6 + 0x111f6d)
                                                       #1  0x00007ffa4cf8c32d n/a (liblttng-ust.so.0 + 0x1b32d)
                                                       #2  0x00007ffa51310450 start_thread (libpthread.so.0 + 0x9450)
                                                       #3  0x00007ffa516c1d53 __clone (libc.so.6 + 0x117d53)
                                                       
                                                       Stack trace of thread 2825:
                                                       #0  0x00007ffa516bbf6d syscall (libc.so.6 + 0x111f6d)
                                                       #1  0x00007ffa4cf8c32d n/a (liblttng-ust.so.0 + 0x1b32d)
                                                       #2  0x00007ffa51310450 start_thread (libpthread.so.0 + 0x9450)
                                                       #3  0x00007ffa516c1d53 __clone (libc.so.6 + 0x117d53)
```

**Desktop**
 - Distro: Ubuntu
 - Desktop Environment: XFCE, Blueman
 - Version: 21.04

**Comment by souravdas142** [comment]
Thanks for the report.

can you please also report to the upstram : https://gitlab.freedesktop.org/pipewire/pipewire/-/issues

Recently they have pushed many libcamera related fixes. May be they have missed something to fix.

**Comment by souravdas142** [comment]
Any leads ? Please report to the devs. I don't have a machine currently running on any version of ubuntu or the same DE. So I can't reproduce it. Its you and devs (by interacting) who can solve the problem for others. It also may have the problem with `libcamera` itself or the `Gstreamer` ? But I think there is a high chance to get solved by reporting to the devs.

**Comment by souravdas142** [comment]
does it fix in 0.3.38 ?

**Comment by alexzeitgeist** [comment]
Hi. Sorry, couldn't reply before weekend.

Strangely, the error disappeared with the second and subsequent reboots. Then, when updated to 0.3.38, and with the first reboot, the error reappeared. And again - after that, with the next reboot, the error disappeared. I am not familiar with GStreamer and gst-plugin-scan, but it seems to fix itself.
